SourceryFramework/Sources/Parsing/FileParser.swift

```diff
@@ -893,6 +913,16 @@ extension FileParser {
             return nil
         }
     }
+
+    private func extractComposedTypeNames(from value: String) -> [TypeName]? {
+        guard case let components = value.components(separatedBy: CharacterSet(charactersIn: "&")),
+            components.count > 1 else { return nil }
+
+        let suffixes = components.map { source in
```

Comment:
[nit] this logic seem to belong only to typealias case, when in inheritance or in associated type there will be `:` instead of `=` (probably won't be needed to parse it as SourceKit should already drop it). So maybe for performance considerations we should move this out to the place where we use this function to parse typealises only?

Reply:
thanks done